packages/nuxt/src/app/composables/loading-indicator.ts

```diff
@@ -0,0 +1,135 @@
+import { computed, getCurrentScope, onScopeDispose, ref } from 'vue'
+import type { Ref } from 'vue'
+import { useNuxtApp } from '#app/nuxt'
+
+export type LoadingIndicatorOpts = {
+  /** @default 2000 */
+  duration: number
+  /** @default 200 */
+  throttle: number
+}
+
+function _increase (progress: Ref<number>, num: number) {
+  progress.value = Math.min(100, progress.value + num)
+}
+
+function _hide (isLoading: Ref<boolean>, progress: Ref<number>) {
+  if (import.meta.client) {
+    setTimeout(() => {
+      isLoading.value = false
+      setTimeout(() => { progress.value = 0 }, 400)
+    }, 500)
+  }
+}
+
+export type LoadingIndicator = {
+  _cleanup: () => void
+  progress: Ref<number>
+  isLoading: Ref<boolean>
```

Comment:
Noticed `isLoading` is also reactive is it intended to let users modify it's value? If yes shall we watch?

Reply:
I think we should expose a computed, namely:

`isLoading: computed(() => isLoading.value)`